source/common/config/grpc_mux_impl.cc

```diff
@@ -181,31 +184,37 @@ void GrpcMuxImpl::onDiscoveryResponse(
     // To avoid O(n^2) explosion (e.g. when we have 1000s of EDS watches), we
     // build a map here from resource name to resource and then walk watches_.
     // We have to walk all watches (and need an efficient map as a result) to
-    // ensure we deliver empty config updates when a resource is dropped.
-    std::unordered_map<std::string, ProtobufWkt::Any> resources;
-    SubscriptionCallbacks& callbacks = api_state_[type_url].watches_.front()->callbacks_;
+    // ensure we deliver empty config updates when a resource is dropped. We make the map ordered
+    // for test determinism.
+    std::vector<DecodedResourceImplPtr> resources;
+    std::map<std::string, DecodedResourceRef> resource_ref_map;
```

Comment:
nit: theoretically `absl::btree_map` should be faster here and elsewhere.

Reply:
Will fix here, but not elsewhere, since that involves plumbing into some code untouched by this PR, and it seems to add a bit more risk.